enrollment-server-onboarding/src/main/java/com/wultra/app/onboardingserver/impl/service/IdentityVerificationOtpService.java

```diff
@@ -133,6 +155,54 @@ public boolean isUserVerifiedUsingOtp(String processId) {
         return otp.getStatus() == OtpStatus.VERIFIED;
     }
 
+    private OtpVerifyResponse verifyPresenceCheck(final OnboardingProcessEntity process, final OtpVerifyResponse response) throws OnboardingProcessException {
+        final String processId = process.getId();
+        if (!identityVerificationConfig.isPresenceCheckEnabled()) {
+            logger.debug("Presence check is not enabled, Process ID: {}", processId);
+            return response;
+        }
+
+        logger.debug("Evaluating result of presence check, Process ID: {}", processId);
+
+        final IdentityVerificationEntity idVerification = identityVerificationRepository.findFirstByActivationIdOrderByTimestampCreatedDesc(process.getActivationId()).orElseThrow(() ->
+                new OnboardingProcessException("No IdentityVerification found for Process ID: " + processId));
+        final String errorDetail = idVerification.getErrorDetail();
+        final String rejectReason = idVerification.getRejectReason();
+
+        if (StringUtils.isNotBlank(errorDetail) || StringUtils.isNotBlank(rejectReason)) {
```

Comment:
Is it save enough to base detection of _presence check_ just on missing error detail and reject reason or to add another db column?

Reply:
It should work for now. We can improve this functionality in a future pull request.